### PR TITLE
Simplify handling of out-of-bound values `Colormap.__call__`.

### PR DESCRIPTION
**Pull request by anntzer**
Mostly, we don't need to manually handle/clip out of bounds float values if we compute the masks early enough (note that these masks are always computed at some point anyways, so there's no cost of doing that a bit earlier).

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/colors.py [hunk 1]
```diff
@@ -708,28 +708,23 @@ def __call__(self, X, alpha=None, bytes=False):
 
         # Take the bad mask from a masked array, or in all other cases defer
         # np.isnan() to after we have converted to an array.
```
Comment (greglucas) [inline lib/matplotlib/colors.py:710]: Move this comment down below with the `mask_bad`

**Review by greglucas — APPROVED** [review]
I agree, this is simpler to follow all of the logic in one spot.

**Review by timhoffm — APPROVED** [review]
Anybody can merge after moving the comment.

**Comment by anntzer** [comment]
done

**Review by dstansby — APPROVED** [review]

**Review by ksunden — APPROVED** [review]

**Pull request merged**